scripts/dats_validator/validator.py

```diff
@@ -105,11 +106,50 @@ def validate_extra_properties(dataset):
                         f"{[k for k in REQUIRED_EXTRA_PROPERTIES.keys()]}")
 
 
+def validate_formats(dataset):
+    """ Checks if the values in the formats field of the JSON object follows the upper case convention without dots. """
+
+    errors_list = []
+    format_exceptions = ['BigWig', 'NIfTI', 'RNA-Seq']
+
+    # check that distributions have a formats property as this is required in the schema
+    for distribution_dict in dataset['distributions']:
+        if 'formats' not in distribution_dict.keys():
+            error_message = f"Validation error in {dataset['title']}: distributions." \
+                            f"formats - 'formats' property is missing under distributions. " \
+                            f"Please add the 'formats' property to 'distributions'."
+            errors_list.append(error_message)
+        else:
+            file_formats_list = reduce(
+                lambda x,y: x+y,
+                [ prop['formats'] for prop in dataset['distributions'] ]
+            )
+            for file_format in file_formats_list:
```

Comment:
Now there is no need for reduce() here, you can access each distribution_dict object and its 'formats' list directly. reduce() returns a list of all formats across all distribution_dicts in one dataset, but we are in a `for` loop for each distribution object. Using reduce() will throw KeyError again if 'formats' is not present in one of them. All that's needed here is:
`else: for file_format in distribution_dict['formats']:` instead of the selected lines.

Reply:
right. Should be resolved in the last commit. I will also need to modify the validation of extraProperties for crawled data as subjects and CONP status cannot be automatically populated by the crawlers... So we need to skip the validation of those two required fields.
Will do that tomorrow